helix-rest/src/test/java/org/apache/helix/rest/metadatastore/accessor/TestZkRoutingDataWriter.java

```diff
@@ -40,15 +41,35 @@
   private static final String DUMMY_SHARDING_KEY = "SHARDING_KEY";
   private MetadataStoreRoutingDataWriter _zkRoutingDataWriter;
 
+  // MockWriter is used for testing request forwarding features in non-leader situations
+  class MockWriter extends ZkRoutingDataWriter {
+    HttpUriRequest calledRequest;
+
+    MockWriter(String namespace, String zkAddress) {
+      super(namespace, zkAddress);
+    }
+
+    // This method does not call super() because the http call should not be actually made
+    @Override
```

Comment:
Shouldn't you make the HTTP call? Otherwise you won't be covering distributed leader election? I think we **should** make the HTTP call here to the leader. Otherwise it's not a true integration test.

Or is this meant to be a unit test?

Reply:
I think you got the wrong file. :) 
The `MockZkRoutingDataWriter` for integration testing is in `MockMetadataStoreDirectoryAccessor.java`, and that's where the actual HTTP calls are made to the leader. This file is for unit-testing, where only the forwarding logic is tested.

Reply:
Ok, so a unit test.